clients/src/test/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumerTest.java

```diff
@@ -87,10 +102,58 @@ public void testSuccessfulStartupShutdown() {
     }
 
     @Test
-    public void testCommitAsync() {
+    public void testInvalidGroupId() {
+        this.groupId = null;
         consumer = newConsumer(time, new StringDeserializer(), new StringDeserializer());
-        consumer.commitAsync();
-        verify(eventHandler).add(any());
+        assertThrows(InvalidGroupIdException.class, () -> consumer.committed(new HashSet<>()));
+    }
+
+    @Test
+    public void testCommitAsync_NullCallback() throws InterruptedException {
+        CompletableFuture<Void> future = new CompletableFuture<>();
+        Map<TopicPartition, OffsetAndMetadata> offsets = new HashMap<>();
+        offsets.put(new TopicPartition("my-topic", 0), new OffsetAndMetadata(100L));
+        offsets.put(new TopicPartition("my-topic", 1), new OffsetAndMetadata(200L));
+
+        PrototypeAsyncConsumer<?, ?> mockedConsumer = spy(newConsumer(time, new StringDeserializer(), new StringDeserializer()));
+        doReturn(future).when(mockedConsumer).commit(offsets);
+        mockedConsumer.commitAsync(offsets, null);
+        future.complete(null);
+        TestUtils.waitForCondition(() -> future.isDone(),
+                2000,
+                "commit future should complete");
+
+        assertFalse(future.isCompletedExceptionally());
+    }
+
+
+    @Test
+    public void testCommitAsync_UserSuppliedCallback() {
```

Comment:
@guozhangwang  - This test and the above are to address your comment about null DefaultCommitCallback.